processing/src/main/java/org/apache/druid/query/rowsandcols/ArrayListRowsAndColumns.java

```diff
@@ -153,11 +154,28 @@ public Column findColumn(String name)
       return new LimitedColumn(retVal, startOffset, endOffset);
     }
 
-    final Function<RowType, Object> adapterForValue = rowAdapter.columnFunction(name);
     final Optional<ColumnType> maybeColumnType = rowSignature.getColumnType(name);
     final ColumnType columnType = maybeColumnType.orElse(ColumnType.UNKNOWN_COMPLEX);
     final Comparator<Object> comparator = Comparator.nullsFirst(columnType.getStrategy());
 
+    final Function<RowType, Object> adapterForValue;
+    if (columnType.equals(ColumnType.STRING)) {
+      // special handling to reject MVDs
+      adapterForValue = f -> {
+        Object value = rowAdapter.columnFunction(name).apply(f);
+        if (value instanceof List) {
+          throw InvalidInput.exception(
+              "Encountered a multi value column [%s]. Window processing does not support MVDs. "
```

Comment:
@sreemanamala How does this code know that a window query is being run? Won't the same error be thrown for non-window queries as well that go through this flow? This class doesn't seem window specific to me. Thoughts?

Reply:
This piece of code is only invoked by window functions. Moreover if columnType is STRING and we get the String comparator. This would fail in compare if the physical value turns out to be a list. 

IMO, if we want to support MVDs, then we need to find a way to unnest the MVDs like the group by query does

Reply:
>IMO, if we want to support MVDs, then we need to find a way to unnest the MVDs like the group by query does

please no, we have `UNNEST`, which you can use with mvds by wrapping with `MV_TO_ARRAY`. people should use that if they want to operate on individual string elements of mvd.

Reply:
@sreemanamala My concern is that this code change would end up returning a window-function specific error message `Window processing does not support MVDs.` even for queries that don't have a window function, if they go through this flow?

> This piece of code is only invoked by window functions.

But the class is supposed/designed to be used generally, I think..

Reply:
I can't really speak to plans about wider use in the future, but today `RowsAndColumns` is not used at all outside of window queries.